pylint/checkers/variables.py

```diff
@@ -1321,7 +1317,16 @@ def _check_consumer(
         current_consumer: NamesConsumer,
         consumer_level: int,
         base_scope_type: Any,
-    ) -> Tuple[VariableVisitConsumerAction, Optional[Any]]:
+    ) -> Union[
+        Tuple[
+            Union[
+                Literal[VariableVisitConsumerAction.CONTINUE],
+                Literal[VariableVisitConsumerAction.RETURN],
+            ],
+            None,
+        ],
+        Tuple[Literal[VariableVisitConsumerAction.CONSUME], List[nodes.NodeNG]],
+    ]:
```

Comment:
```suggestion
    ) ->Tuple[VariableVisitConsumerAction, Optional[List[nodes.NodeNG]]]:
```
Would this be enough ? I know the other one is more specific but it's also very verbose.

Reply:
We could but then we'll need the assert again. The current version because if `CONSUME` is the first value then the second is restricted to be a `list`. That's why `pylance` infers but `mypy` fails to do.